hyperspy/drawing/marker.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2011 The Hyperspy developers
+#
+# This file is part of  Hyperspy.
+#
+#  Hyperspy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  Hyperspy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  Hyperspy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+
+
+class Marker(object):
+
+    """Marker that can be added to the signal figure
+
+    Attributes
+    ----------
+    marker_type : {'line', 'axvline', 'axhline', 'text', 'pointer', 'rect'}
+        Select the type of markers
+    orientation : {None,'v','h'}
+        Orientation for lines. 'v' is vertical, 'h' is horizontal.
+    marker_properties : dictionary
+        Accepts a dictionary of valid (i.e. recognized by mpl.plot)
+        containing valid line properties. In addition it understands
+        the keyword `type` that can take the following values:
+        {'line', 'text'}
+
+    Methods
+    -------
+    set_marker_properties
+        Enables setting the line_properties attribute using keyword
+        arguments.
+    set_data
+        Set the data in a structured array. Each field of data should have
+        the same dimensions than the nagivation axes. Some fields need to be
+        defined depending on the type.
+        For 'line': 'x1','y1','x2','y2' (All of them if orientation
+            is None)
+        For 'axvline': 'x1'
+        For 'axhline': 'y1'
+        For 'text': 'x1','y1','text'
+        For 'pointer': 'x1','y1','size'. 'size' is optional
+        For 'rect': 'x1','y1','x2','y2'
+    add_data
+        Add data to the structured array. Same field as set_data
```

Comment:
I find that the difference between `set_data` and `add_data` is not very clear in the docs. How about adding a couple of words in the `set_data` description that it always overwrites all previously save results? And explaining that `add_data` will only change fields that are explicitly specified by the user?

I may be wrong though, it's up to you :)